security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -335,9 +335,9 @@ func (sc *SecretManagerClient) addFileWatcher(file string, resourceName string)
 	// File is not being watched, start watching now and trigger key push.
 	cacheLog.Infof("adding watcher for file certificate %s", file)
 	if err := sc.certWatcher.Add(file); err != nil {
-		cacheLog.Errorf("%v: error adding watcher for file, skipping watches [%s] %v", resourceName, file, err)
 		numFileWatcherFailures.Increment()
-		return
+		// This can happen if we have hit the limit on max_user_watches on the Node.
```

Comment:
is this a transient issue? I am worried about temporary issues now causing panic.

Or, someone applying a bunch of destination rules referencing certs and crashing all pods in the cluster. That would be pretty bad scenario

Reply:
This is not a transient issue. Once we fail to add a watcher to cert, the cert would never be reloaded. Only way to restore the pod would be restart.

I hear your point about too many DRs causing all pods to crash.  In our scenario we have seen this happen for ingress gateway client and server certs - So all calls going out of ingress are failing with cert expiry message because the cert will not be reloaded. One way to avoid that situation would be to only panic on proxy certs and leave DR certs - but we do not have a way to distinguish that here - but we can some how embed that it is a DR cert in the resourcename.

Reply:
Or should we add retry logic that tries to add filewatchers instead of giving up with an error/panicing after one failure? This may be more reliable it would be able to add watchers when the watchers get free?

Reply:
I don't really know. I don't like a panic, but it may be FUD than actual requirement. It is super worrying that some transient error could blow up the proxy. Additionally, panic will not actually fix the problem most likely, since the container will restart in place and not be moved to a different node?

I don't know a good solution